src/sentry/static/sentry/app/components/dataExport.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+
+import {Client} from 'app/api';
+import Feature from 'app/components/acl/feature';
+import Tooltip from 'app/components/tooltip';
+import {t} from 'app/locale';
+import {Organization} from 'app/types';
+import withApi from 'app/utils/withApi';
+import withOrganization from 'app/utils/withOrganization';
+
+type DataExportPayload = {
+  queryType: number;
+  queryInfo: any; // TODO(ts): Formalize different possible payloads
+};
+
+type Props = {
+  api: Client;
+  organization: Organization;
+  payload: DataExportPayload;
+};
+
+type State = {
+  inProgress: boolean;
+  dataExportId?: number;
+};
+
+const TooltipMessages = {
+  start: "We'll get all your data in one place and email you when it's ready",
+  progress: "We'll email you when it's ready",
+} as const;
+
+class DataExport extends React.Component<Props, State> {
+  state: State = {
+    inProgress: false,
+  };
+
+  async startDataExport() {
+    const {
+      api,
+      organization: {slug},
+      payload: {queryType, queryInfo},
+    } = this.props;
+    const {id: dataExportId} = await api.requestPromise(
+      `/organizations/${slug}/data-export/`,
+      {
+        method: 'POST',
+        data: {
+          query_type: queryType,
+          query_info: queryInfo,
+        },
```

Comment:
This is okay right? The backend should _receive_ `snake_case` and the frontend should be using `camelCase` all the way until it fires off the request.

Reply:
It looks ok to me.